File: InJava/app/src/main/java/com/barryzeha/pomodoroapp/model/TaskModel.java
package com.barryzeha.pomodoroapp.model;

import androidx.room.Entity;
import androidx.room.PrimaryKey;

import java.util.Objects;

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 18/10/2022
 * Copyright (c)  All rights reserved.
 ***/
@Entity(tableName = "task_table")
public class TaskModel {
    @PrimaryKey(autoGenerate = true) int id;
    String taskName;
    long initTaskTimestamp;
    long endTaskTimestamp;
    int totalCycles;
    long totalTime;

    public int getId() {
        return id;
    }

    public void setId(int id) {
        this.id = id;
    }

    public String getTaskName() {
        return taskName;
    }

    public void setTaskName(String taskName) {
        this.taskName = taskName;
    }

    public long getInitTaskTimestamp() {
        return initTaskTimestamp;
    }

    public void setInitTaskTimestamp(long initTaskTimestamp) {
        this.initTaskTimestamp = initTaskTimestamp;
    }

    public long getEndTaskTimestamp() {
        return endTaskTimestamp;
    }

    public void setEndTaskTimestamp(long endTaskTimestamp) {
        this.endTaskTimestamp = endTaskTimestamp;
    }

    public int getTotalCycles() {
        return totalCycles;
    }

    public void setTotalCycles(int totalCycles) {
        this.totalCycles = totalCycles;
    }

    public long getTotalTime() {
        return totalTime;
    }

    public void setTotalTime(long totalTime) {
        this.totalTime = totalTime;
    }

    @Override
    public boolean equals(Object o) {
        if (this == o) return true;
        if (o == null || getClass() != o.getClass()) return false;
        TaskModel taskModel = (TaskModel) o;
        return id == taskModel.id;
    }

    @Override
    public int hashCode() {
        return Objects.hash(id);
    }
}

File: InKotlin/app/src/main/java/com/barryzeha/pomodoroapp/model/TaskModel.kt
package com.barryzeha.pomodoroapp.model

import androidx.room.Entity
import androidx.room.PrimaryKey

/****
 * Project PomodoroApp
 * Created by Barry Zea H. on 04/10/2022
 * Copyright (c)  All rights reserved.
 ***/
@Entity(tableName = "task_table")
data class TaskModel(
    @PrimaryKey(autoGenerate = true) val id:Int=0,
    var taskName:String="",
    var initTaskTimestamp:Long=0L,
    var endTaskTimestamp:Long=0L,
    var totalCycles:Int=0,
    var totalTime:Long=0L
    ) {
    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false

        other as TaskModel

        if (id != other.id) return false

        return true
    }

    override fun hashCode(): Int {
        return id
    }
}
